src/main/java/com/redislabs/redistimeseries/RedisTimeSeries.java

```diff
@@ -143,27 +122,32 @@ public boolean create(String key, long retentionTime, Map<String, String> labels
    */
   public boolean create(String key, long retentionTime, boolean uncompressed, Map<String, String> labels){
     try (Jedis conn = getConnection()) {
+      byte[][] args = tsCreateArgs(key, retentionTime, uncompressed, labels);
+      return sendCommand(conn, Command.CREATE, args).getStatusCodeReply().equals("OK");
+    }
+  }
 
-      byte[][] args = new byte[3 + (labels==null ? 0 : 2*labels.size()+1) + (uncompressed?1:0)][];
-      int i=0;
+  private static byte[][] tsCreateArgs(String key, long retentionTime, boolean uncompressed, Map<String, String> labels) {
+    byte[][] args = new byte[1 + (labels==null ? 0 : 2*labels.size()+1) + (retentionTime>=0 ? 2 : 0 ) + (uncompressed?1:0)][];
+    int i=0;
 
-      args[i++] = SafeEncoder.encode(key);
+    args[i++] = SafeEncoder.encode(key);
+    if(retentionTime>=0) {
```

Comment:
this is a behavior change and will hide user's mistakes (giving negative numbers)

Reply:
It falls in the same premise as this one:
```timestamp>0 ? Protocol.toByteArray(timestamp) : STAR```. Adding a private method to accept Long